client/etcd/client.go

```diff
@@ -97,11 +101,17 @@ type csclient struct {
 }
 
 func (c *csclient) Services() (services.Services, error) {
-	c.createServices()
+	c.sdOnce.Do(func() {
+		c.sd, c.sdErr = c.createServices(services.NewOptions())
```

Comment:
Perhaps rename `Services` to `DefaultServices` and `sd` and `sdErr` to `defaultServices` and `defaultServicesErr`?